### DISCUSSION TITLE
"Tab completion" feature somehow leaves new lines and awkward values

### DISCUSSION THREAD

**Discussion by skatkov** [ghostty-org/ghostty#2741] [Q&A]
When I'm traversing through directories in ghostty terminal, using “tab completion” features gives me a weird artifact. It shows me possible values for completion, I start typing to narrow down suggestions and then pick one. After I pick one of the values, the prompt jumps to a new line, leaving previous value I typed, without filling out the selected value.

Here is a screenshot.
![Image](https://github.com/user-attachments/assets/9a5f0171-c6d9-4a26-9f14-c7d145b93823)

This issue is not always reproducible, in some cases ghostty snaps out of it and works as expected.
![Image](https://github.com/user-attachments/assets/32b2776f-4bda-4917-b0fc-964be746a089)

I'm using:
- Manjaro
- Zsh

**Comment by 00-kat** [comment]
*Incredibly* late, but does `shell-integration = none` change anything?
